### PR TITLE
Adding the option to customize Credentials creation

### PR DESCRIPTION
**Pull request by davidrabinowitz**

### CODE REVIEW

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java [hunk 1]
```diff
@@ -297,4 +299,50 @@ static Field.Mode nullableIfNull(Field.Mode mode) {
...
+   * class
+   */
+  public static <T> T createVerifiedInstance(
+      String fullyQualifiedClassName, Class<T> requiredClass) {
+    try {
+      Class<?> clazz = Class.forName(fullyQualifiedClassName);
+      Object result = clazz.newInstance();
+      if (!requiredClass.isInstance(result)) {
+        throw new IllegalArgumentException(
+            String.format(
```
[the remark this answers is not in the document]
Comment (davidrabinowitz) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java:314]: @sonatype-lift ignore

**Review by VinayLondhe14 — APPROVED** [review]

**Review by VinayLondhe14 — APPROVED** [review]

**Pull request merged**